perceptia-one/perceptia/src/components/Question.js

```diff
@@ -15,37 +15,39 @@ import { withStyles } from '@material-ui/core/styles';
 import QuizInfo from './QuizInfo';
 import { BrowserRouter as Router, Route, Link } from "react-router-dom";
 import { DeleteForever, Edit, SaveOutlined, NotInterested, NavigateNext } from "@material-ui/icons";
-import "./quiz.css"
+import "./quiz.css";
 import constants from "./constants";
 import CircularProgress from '@material-ui/core/CircularProgress';
 
 class Question extends React.Component {
 
     constructor(props) {
         super(props);
+        console.log(props);
         this.state = {
             edit: false,
-            question:"",
-            question_type:null,
-            questionId:"",
-            answerOptions:[],
-            newText:"",
-            newAnswer:"",
+            question: props.question,
+            question_type: +props.question_type,
+            questionId: props.questionId,
+            answerOptions:props.answerOptions,
+            newText: props.question,
+            newAnswer: "",
             busy: false,
-        }
+        };
     }
 
-    static getDerivedStateFromProps(nextProps) {
+    /*static getDerivedStateFromProps(nextProps) {
+        console.log(nextProps)
         return {
             question:nextProps.question,
-            newText:nextProps.question,
+            
             question_type:+nextProps.question_type,
             questionId:nextProps.questionId,
             answerOptions:nextProps.answerOptions,
-            newAnswer:"",
+            
             busy: false,
-        }
-    }
+        };
+    }*/
 
     deleteQuestion(){
```

Comment:
I read the article you provided (I see it says 2016, so I went looking, mainly because it said it was a stage-2 feature at the time of writing). I found this https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1 which seems to suggest using the "property arrow functions" increase space complexity and decrease Ops/s.

I'm of the mind that having it simpler to read and maintain is better at this time. It's worth reading more about, but we can follow up on it more offline.

Reply:
Thanks for looking into that. I don't remember where I read about it, but it only mentioned the performance hit of using the arrow function in the render function to bind this, but it looks like its a performance hit wherever it's done, thanks!